### ISSUE TITLE
passing additional parameters to ClickHouse connection

### ISSUE THREAD

**Issue by ne1r0n** [whisklabs/airflow-clickhouse-plugin#16]
It would be great to be able to specify additional parameters for ClickHouse driver, for example, send_receive_timeout, client_name, etc..

As a workaround, I hardcoded the necessary parameters in clickhouse_hook.py.

Do you have any idea how to implement this by correct pythonyc way?

**Comment by bryzgaloff** [comment]
Airflow `Connection` [has](https://airflow.apache.org/docs/stable/_api/airflow/models/connection/index.html#airflow.models.connection.Connection.extra) `extra` attribute which should store these additional parameters. They can be set through Airflow web UI through Admin -> Connections -> Edit/Create Connection via `extra` field.

A proper way is to add contents of `conn.extra` property to `self.create_connection` call in `ClickHouseHook.get_conn` [method](https://github.com/whisklabs/airflow-clickhouse-plugin/blob/master/airflow_clickhouse_plugin/hooks/clickhouse_hook.py#L20). But please specify constructor arguments explicitly. Like this:
```python
extra_connection_kwargs = {}
if 'send_receive_timeout' in conn.extra:
    extra_connection_kwargs['send_receive_timeout'] = conn.extra['send_receive_timeout']
# add other expected extra arguments handling here
return self.create_connection(…, **extra_connection_kwargs)
```

And not like `self.create_connection(…, **conn.extra)`, since this can cause unexpected behaviour  if any extra property overrides one of the required arguments. Using an additional `extra_constructor_kwargs` dict allows us to control the keys which can be added through `conn.extra` without overriding default values if the property is not present. For instance, `send_receive_timeout` [defaults to 300](https://clickhouse-driver.readthedocs.io/en/latest/api.html#connection) and we should not preserve this behaviour.

**Comment by bryzgaloff** [comment]
@ne1r0n could you please create and test a PR with the proposed solution? Thank you in advance.

**Comment by bryzgaloff** [comment]
Released as [0.5.6](https://pypi.org/project/airflow-clickhouse-plugin/0.5.6/).
